### PR TITLE
Social sharing stuff

### PR DESCRIPTION
**Pull request by DrShpongle**
Added:
- Tweet button
- Copy to Clipboard button

ezgif com-optimize

Image 2020-09-24 at 1 44 55 PM

### CODE REVIEW

File: src/components/pages/lessons/LessonInfo.tsx [hunk 1]
```diff
@@ -47,6 +48,71 @@ const NextUp: FunctionComponent<NextUpProps> = ({data, currentLessonSlug}) => {
...
+  if (source) {
+    return (
+      <a
+        className={`flex items-center rounded p-2 bg-gray-200 hover:bg-gray-400 transition-colors ease-in-out duration-150 ${className}`}
+        target="_blank"
+        rel="noopener noreferrer"
+        href={`https://twitter.com/intent/tweet/?text=${encodeURIComponent(
+          source.title + ', lesson by @eggheadio',
+        )}&url=${encodeURIComponent(
+          `https://next.egghead.io/lessons/${source.slug}/`,
```
Comment (vojtaholik) [inline src/components/pages/lessons/LessonInfo.tsx:67]: use `NEXT_PUBLIC_REDIRECT_URI` or similar from ENV instead of hardcoding it?

Comment (joelhooks) [reply src/components/pages/lessons/LessonInfo.tsx:67]: For same domain if you are URL building `/` works

Does source have a `path`?

Comment (DrShpongle) [reply src/components/pages/lessons/LessonInfo.tsx:67]: yes, it has

Comment (DrShpongle) [reply src/components/pages/lessons/LessonInfo.tsx]: but seems we need `process.env.NEXT_PUBLIC_REDIRECT_URI` anyway

File: src/components/pages/lessons/LessonInfo.tsx [hunk 2]
```diff
@@ -47,6 +48,66 @@ const NextUp: FunctionComponent<NextUpProps> = ({data, currentLessonSlug}) => {
...
+  path: string
+  className?: string
+}> = ({title, path, className = ''}) => {
+  return title && path ? (
+    <a
+      className={`flex items-center rounded p-2 bg-gray-200 hover:bg-gray-400 transition-colors ease-in-out duration-150 ${className}`}
+      target="_blank"
+      rel="noopener noreferrer"
+      href={`https://twitter.com/intent/tweet/?text=${encodeURIComponent(
+        title + ', lesson by @eggheadio',
```
Comment (vojtaholik) [inline src/components/pages/lessons/LessonInfo.tsx:62]: can be more specific → "lesson by @instructorHandle on @egghead.io"

**Review by vojtaholik — APPROVED** [review]

**Pull request merged**